pkg/3scale/amp/component/system.go

```diff
@@ -150,6 +151,11 @@ func (system *System) getSystemSMTPEnvsFromSMTPSecret() []v1.EnvVar {
 		helper.EnvVarFromSecret("SMTP_OPENSSL_VERIFY_MODE", SystemSecretSystemSMTPSecretName, SystemSecretSystemSMTPOpenSSLVerifyModeFieldName),
 	}
 
+	if system.Options.SmtpSecretOptions.FromAddress != nil &&
+		*system.Options.SmtpSecretOptions.FromAddress != "" {
```

Comment:
The description of the PR says

> If the field is set in the secret then the `NOREPLY_EMAIL` is sent to system, even if its value is the empty string

but it looks like it won't set the env var if the value in the secret is the empty string after all.

Reply:
hi @guicassolato ,

Sorry, that initial comment is outdated :( . See https://github.com/3scale/3scale-operator/pull/566#discussion_r561932141

Reply:
What's implemented is better than what's described anyway 🙂 Nice job!